source/plugin/data/datamanipulators.rst

```diff
@@ -0,0 +1,111 @@
+=================
+Data Manipulators
+=================
+
+Accessing and modifying data
+============================
+
+A data manipulator represents a certain component and all of its data. It stores a representation of that data and can
+be offered to or created from data holders which possess a matching component. Again, let's use an example. And again
+try to heal someone (or something).
+
+**Code Example: Healing with data manipulators**
+
+.. code-block:: java
+
+    public static DataTransactionResult heal(DataHolder target) {
+        Optional<HealthData> healthOptional = target.getOrCreate(HealthData.class);
+        if (healthOptional.isPresent()) {
+            HealthData healthData = healthOptional.get();
+
+            double maxHealth = healthData.maxHealth().get();
+            MutableBoundedValue<Double> currentHealth = healthData.health();
+            currentHealth.set(maxHealth);
+            healthData.set(currentHealth);
+
+            target.offer(healthData);
+        }
+    }
+
+First we need to check if our target has health data. We do so by first asking it to provide us with its health
+data by passing its class to the ``getOrCreate()`` method. We get an ``Optional`` which we can use for our check.
+If the target does not support health data, it will be absent. But if the health data is present, it now contains
+a mutable copy of the data present on the data holder. We make our alterations and finally offer the changed data
+back to our target, where it is accepted (again, ``offer()`` will return a ``DataTransactionResult`` which we will
+just discard here).
```

Comment:
Not discard, but rather, disregarding. Even though, sometimes it should be checked for any possible errors.